### PR TITLE
Kernel solve improvements

### PR DESCRIPTION
**Pull request by Matt-Ord**

### CODE REVIEW

File: surface_potential_analysis/kernel/solve.py [hunk 1]
```diff
@@ -468,7 +590,7 @@ def get_noise_operators_real_isotropic_stacked_taylor_expansion(
     ).astype(np.complex128)
     operator_coefficients *= n_states
 
-    operators = get_operators_for_real_isotropic_noise(basis_x, n_terms=n + 1)
+    operators = get_operators_for_real_isotropic_noise(basis_x, n=n + 1)
 
     return {
         "basis": operators["basis"],
```
Comment (Winnie08L) [inline surface_potential_analysis/kernel/solve.py:596]: make code work for system of arbitrary dimension, seems good to me, have they been tested and do they work?

File: surface_potential_analysis/util/plot.py [hunk 2]
```diff
@@ -378,10 +378,13 @@ def plot_data_2d_k(
     -------
     tuple[Figure, Axes, QuadMesh]
     """
-    idx = get_max_idx(basis, data, axes) if idx is None else idx
+    basis_k = stacked_basis_as_fundamental_momentum_basis(basis)
+    converted_data = convert_vector(data, basis, basis_k)
 
-    coordinates = get_k_coordinates_in_axes(basis, axes, idx)
-    data_in_axis = get_data_in_axes(data.reshape(basis.shape), axes, idx)
+    idx = get_max_idx(basis_k, data, axes) if idx is None else idx
```
Comment (Winnie08L) [inline surface_potential_analysis/util/plot.py:384]: data should be converted_data?

Comment (Matt-Ord) [reply surface_potential_analysis/util/plot.py]: Very good spot!

File: surface_potential_analysis/kernel/solve.py (hunk 1 above)
Comment (Matt-Ord) [reply surface_potential_analysis/kernel/solve.py:596]: So it doesn't actually work - but I haven't broken 1d. This was mostly about making the code cleaner in preparation, and building out some visualizations that work in 2d

**Pull request merged**